### ISSUE TITLE
Request: Add full drivers support

### ISSUE THREAD

**Issue by TheCzechTayen** [pop-os/cosmic-epoch#1233]
Make a repository like **cosmic-drivers** and put all available linux drivers there. Using autodetect, the system would download and install only the necessary drivers for the given pc.

Almost every distribution has some kind of message like missing drivers, missing firmware etc. It would be great to have everything in one place like mouses/keyboards/firmware/cpu/gpu etc (literally every driver which exist for linux) and possibly add the **Drivers** page to cosmic-settings

**Comment by git-f0x** [comment]
This is not something that's related to a desktop environment.
Most drivers are built into the kernel, and I haven't seen such messages in any distro, unless you're referring to the "Possible missing firmware for..." messages in the terminal during kernel upgrades, which are irrelevant.
The things that mostly need separately installed drivers are Nvidia GPUs (for now) and some printers (a lot are driverless), and both can be easily installed or are automatically installed in most distros (if the vendor provides drivers).
Firmware updates are handled with `fwupd`.

**Comment by TheCzechTayen** [comment]
Yea but I still miss the drivers repo. There is no good place in the Linux world yet and drivers are scattered everywhere. The cosmic devs are mega determined and I think they could handle this. Anyway, having a drivers page in the settings is in my opinion an advantage anyway, rather than having other apps for it.

If they unified these things including the drivers page, maybe even the fancontrol page, or overclocking, and god knows what else directly in cosmic-settings, there wouldn't be so many apps that don't even work properly. Just my suggestion.

**Comment by git-f0x** [comment]
But this isn't at all related to a desktop environment. Drivers are distro-specific, so it's up to individual distros to implement something like this.
Drivers aren't scattered everywhere, they're in the kernel, official dkms drivers that are easily/automatically installed (e.g. Nvidia), or unofficial 3rd-party drivers for devices from vendors that don't support Linux, which may or may not be available in your distro's repo (so you would have to compile them).
Regardless, you usually don't have to install any drivers, other than what I already mentioned (which are often automatically installed).
There will be a firmware page in Settings for updating the firmware of devices that support `fwupd`.
Overclocking, fan-control, etc. are very out of scope for a settings app, and no OS has such things in Settings (especially overclocking).

**Comment by TheCzechTayen** [comment]
Fine then. How i said, it was just a suggestion. If its a bad suggestion, sorry then.

That "possibly missing firmware" is pointless then? Maybe i just dont understand it correctly

**Comment by git-f0x** [comment]
If possible, a list of devices page might be added to Settings at some point, with maybe an option to install drivers if there could somehow be a driver repo for things not included in the kernel. But that would require distro support, so it isn't very simple. And definitely won't be there for the first release.
>That "possibly missing firmware" is pointless then? 

That's just a message displaying AMD gpu firmware that isn't installed on your system, because you don't use that GPU (e.g. if you have an RX 5700XT in your system, firmware for an RX 7900XTX isn't used).

**Comment by TheCzechTayen** [comment]
Ah understand. So this messages are not related with drivers. i tried solved it every time. I thought I was missing drivers or something was not set up properly, thanks for the clarification. In that case, my suggestion is not needed

**Comment by git-f0x** [comment]
Yeah, for GPU drivers, you shouldn't need to do anything after installing the OS, if you have AMD or Intel graphics (Nvidia as well, but the drivers built into the kernel aren't performant with modern GPUs, since they aren't developed by Nvidia).
For Nvidia, it's usually easy to install them, and in the future, they'll likely also switch to using the kernel driver, with an option of using an open-source userspace driver (developed by the community), or their proprietary one with support for CUDA, etc.

**Comment by TheCzechTayen** [comment]
Im AMD only guy, both CPU and GPU. i dont care about intel or nvidia at all

**Comment by acolombier** [comment]
IMOH, I think it is a fair suggestion for `cosmic-settings` to have a driver page, and for PopOS distro to try and smooth the driver experience (auto detection and management) . Agreed with @git-f0x that this is out of scope as such for Cosmic Epoch, but also agree with @TheCzechTayen that it would be great to make an effort and streamlining the process of handling **any** drivers and not require our user to be tech savvy to get their system in a working state. It certainly has been a global issue with Linux distros, and seeing PopOS trying to lead the way with smoothing that would be great for sure.

**Comment by TheCzechTayen** [comment]
Yes, that's what I wanted, a lot of distros have driver managers or some other apps for system management, fan control and similar things, which are often half-functional or hard to understand. Having most of the settings in cosmic-settings and not having to dig through several programs and do trial/error would be great.

**Comment by emanuc** [comment]
Maybe this is something that should be included in the Cosmic Store? Perhaps for consistency with what other desktop environments do, such as [GNOME](https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/2034) (gnome-software) and [KDE Plasma ](https://invent.kde.org/plasma/discover/-/merge_requests/1048)(plasma-discover), which have a dedicated “hardware” or “codec” section in their store.

<img width="939" height="732" alt="Image" src="https://github.com/user-attachments/assets/2fcf09fb-aacc-4ea1-8db9-fd20d3ce6d14" />

<img width="769" height="922" alt="Image" src="https://github.com/user-attachments/assets/32bc143d-c042-4b5e-afe2-680b5934b5d9" />

<img width="655" height="808" alt="Image" src="https://github.com/user-attachments/assets/5c65f863-ed53-4c70-b8d6-435973243838" />

**Comment by leviport** [comment]
A driver manager as part of cosmic-store is a feature request being tracked here: https://github.com/pop-os/cosmic-store/issues/419
